nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/FileSystemRepository.java

```diff
@@ -392,8 +399,11 @@ public long getContainerUsableSpace(String containerName) throws IOException {
         if (path == null) {
             throw new IllegalArgumentException("No container exists with name " + containerName);
         }
-
-        return Files.getFileStore(path).getUsableSpace();
+        long usableSpace=path.toFile().getUsableSpace();
+        if(usableSpace==0) {
+            throw new RuntimeException("System returned usable space of the partition for " + containerName + " is zero byte. Nifi can not create a zero sized FileSystemRepository");
```

Comment:
We shouldn't be throwing an Exception here at all. If the content repository runs out of disk space, it will return 0. This is a very valid value and should not result in an Exception being thrown.